src/aleph/netty.clj

```diff
@@ -725,63 +728,192 @@
 
 ;;;
 
+(defn coerce-ssl-provider [provider]
+  (case provider
+    :jdk SslProvider/JDK
+    :openssl SslProvider/OPENSSL
+    :openssl-refcnt SslProvider/OPENSSL_REFCNT))
+
+(set! *warn-on-reflection* false)
```

Comment:
This should go at the top, right after the `ns`

Reply:
Wait, I just reread this. It's being set to false temporarily. We should be able to set the necessary hints to avoid reflection.

E.g., something like `^"[Ljava.security.cert.X509Certificate;"` for X509Certificate arrays.

Reply:
Maybe it could be also applied on all namespaces on the project.clj.

Reply:
I would remove all the `(set! *warn-on-reflection* false)`, most of them should be fixable though.

Reply:
Oof, that was trickier than expected :sweat_smile:  All done now. Perhaps there is a less pedestrian way to solve this?

This even turned up a bug: The code was calling `.provider`  while the method is actually called `.sslProvider`.

I also noticed that passing in a  `javax.net.ssl.TrustManager` as `trust-store` was missing, so added that for good measure.

Reply:
> Perhaps there is a less pedestrian way to solve this?

@DerGuteMoritz Generally, Clojure has crappy support for arrays, and Java has crappy support for array types, so... ¯\\\_(ツ)_/¯

One way you can do this is to use a snippet like `(println (str "^\"" (pr-str (class (make-array SomeClass 1))) "\""))` to get the array type hint for SomeClass.

Reply:
Indeed :smile:  I was referring to the whole approach of using `instance?` checks actually which you also commented on below. Resolving this thread then!